### PR TITLE
Allowing modular and custom assertions as dependencies

### LINKED ISSUE #115: modular and custom assertions as dependencies
When they are included as dependencies they should be packaged in the `gw7` package

Modular Assertions should be references like this:
```gradle
dependencies {
    assertion "com.ca.apim.gateway:hello-world:1.2.3"
}
```
Custom Assertions should be references like this:
```gradle
dependencies {
    customassertion "com.ca.apim.gateway:hello-world:1.2.3"
}
```

### PR DESCRIPTION
**Pull request by joao-borges**
Fixes #115 

## Proposed Changes
* Create new types of gradle configurations to allow new dependency types `assertion` and `customassertion`
* Pick up the resolved dependencies and pack into GW7 file in the proper path so it gets automatically loaded into the gateway
* Unit tests

### CODE REVIEW

**Review by apigatewayopensource — COMMENTED** [review]
SonarQube analysis found issues:
Bug Bugs: 0
Vulnerability Vulnerabilities: 0
Code Smell Code Smells: 5

Including the following issue(s) which could not be reported in line:
1. Code Smell Code Smell: Remove this unused import 'java.util.Spliterator'. ([more](https://sonarcloud.io/project/issues?id=com.ca.apim.gateway%3Agateway-developer-plugin&pullRequest=137&issues=AWczk9qV0POp3Yiz_Pc6&open=AWczk9qV0POp3Yiz_Pc6))
1. Code Smell Code Smell: Remove this unused import 'java.util.Spliterators'. ([more](https://sonarcloud.io/project/issues?id=com.ca.apim.gateway%3Agateway-developer-plugin&pullRequest=137&issues=AWczk9qV0POp3Yiz_Pc2&open=AWczk9qV0POp3Yiz_Pc2))
1. Code Smell Code Smell: Remove this unused import 'java.util.function.Function'. ([more](https://sonarcloud.io/project/issues?id=com.ca.apim.gateway%3Agateway-developer-plugin&pullRequest=137&issues=AWczk9qV0POp3Yiz_Pc3&open=AWczk9qV0POp3Yiz_Pc3))
1. Code Smell Code Smell: Remove this unused import 'java.util.stream.Collectors'. ([more](https://sonarcloud.io/project/issues?id=com.ca.apim.gateway%3Agateway-developer-plugin&pullRequest=137&issues=AWczk9qV0POp3Yiz_Pc4&open=AWczk9qV0POp3Yiz_Pc4))

[See all issues in SonarCloud](https://sonarcloud.io/project/issues?id=com.ca.apim.gateway%3Agateway-developer-plugin&pullRequest=137&resolved=false)

File: gateway-developer-plugin/src/main/java/com/ca/apim/gateway/cagatewayconfig/tasks/gw7/Packager.java [hunk 1]
```diff
@@ -23,16 +25,23 @@
 import java.util.stream.Stream;
 import java.util.stream.StreamSupport;
```
Comment (apigatewayopensource) [inline gateway-developer-plugin/src/main/java/com/ca/apim/gateway/cagatewayconfig/tasks/gw7/Packager.java]: Code Smell Code Smell: Remove this unused import 'java.util.stream.StreamSupport'. ([squid:UselessImportCheck](https://sonarcloud.io/organizations/ca-api-gateway/rules#rule_key=squid%3AUselessImportCheck))

[See it in SonarCloud](https://sonarcloud.io/project/issues?id=com.ca.apim.gateway%3Agateway-developer-plugin&pullRequest=137&issues=AWczk9qV0POp3Yiz_Pc5&open=AWczk9qV0POp3Yiz_Pc5)

[ISSUE_KEY]: AWczk9qV0POp3Yiz_Pc5

**Review by yves-chan — COMMENTED** [review]
Just one typo, otherwise looks good to me!

**Review by yves-chan — APPROVED** [review]
Looks great!

**Pull request merged**